cachi2/core/package_managers/bundler/main.py

```diff
@@ -24,5 +32,67 @@ def fetch_bundler_source(request: Request) -> RequestOutput:
 
 def _resolve_bundler_package(package_dir: RootedPath, output_dir: RootedPath) -> list[Component]:
     """Process a request for a single bundler package."""
-    parse_lockfile(package_dir)
+    dependencies = parse_lockfile(package_dir)
+    _get_main_package_name_and_version(package_dir, dependencies)
     return []
+
+
+def _get_main_package_name_and_version(
+    package_dir: RootedPath,
+    dependencies: ParseResult,
+) -> tuple[str, Optional[str]]:
+    """
+    Get main package name and version.
+
+    The main package is the package that is being processed by cachi2.
+    Not any of its dependencies.
+    """
+    name_and_version = _get_name_and_version_from_lockfile(dependencies)
+    if name_and_version is not None:
+        return name_and_version
+
+    log.info("Failed to extract name and version from gemspec file")
+
+    # fallback to origin remote
+    try:
+        name = _get_repo_name_from_origin_remote(package_dir)
+    # if the git repository does not have an origin remote
+    except UnsupportedFeature:
+        raise PackageRejected(
+            reason="Failed to extract package name from origin remote",
+            solution=(
+                "Please specify package name and version in a way that Cachi2 understands,\n"
+                "or make sure that the directory Cachi2 is processing is a git repository with\n"
+                "an 'origin' remote, in which case Cachi2 will infer the package name from the remote URL."
+            ),
+        )
+
+    return name, None
+
+
+def _get_name_and_version_from_lockfile(dependencies: ParseResult) -> Optional[tuple[str, str]]:
+    """
+    Extract package name and version from dependencies in the Gemfile.lock.
+
+    Note that having a gemspec file is an edge case when the package is not an actual gem.
```

Comment:
nitpick: If we say we're extracting name and version from Gemfile.lock then it IMO deserves a small explanation:

_Gemfile.lock only contains name/version information for the package we're processing when a gemspec file is present in the repository. 
Note that ...._